### ISSUE TITLE
Potential bug message with Dogleg

### ISSUE THREAD

**Issue by renato145** [argmin-rs/argmin#246]
Hi, I was trying to optimize the rosenbrock problem from the example of the book using `Dogleg` and got the following message: `Potential bug: "tau is bigger than 2, this is not supposed to happen.". This is potentially a bug.`

This is the code for the optimization part:
```rust
let dogleg = Dogleg::new();
    let dogleg_solver = TrustRegion::new(dogleg);
    let dogleg_res = Executor::new(problem, dogleg_solver)
        .add_observer(SlogLogger::term(), ObserverMode::Always)
        .configure(|state| state.param(init_param).max_iters(iterations))
        .run()
        .unwrap();
```

**Comment by stefan-k** [comment]
Thanks for reporting! I'm having difficulty reproducing this. Could you provide a minimal example, please?

**Comment by renato145** [comment]
Yes, its here: https://gist.github.com/renato145/7c526413f10765f2c6dca53e2d18ca8d

**Comment by stefan-k** [comment]
Thanks a lot for the example! There was something wrong with the calculation of `tau`. I did the math again and now it converges from your starting point without error. The fix is in #247.
